test/abstracts.md

```diff
@@ -0,0 +1,107 @@
+# Abstract
+
+The following content represents ideal steps
+performed by hyperHTML to grant performance,
+recycle everything it can, and update layouts.
+
+It's rather a specification than current implementation,
+but the goal is to optimize such specification as much as possible
+and achieve best performance on top of that.
+
+```
+hyperHTML(statics, ...interpolations)
+│
+└▶  is `statics` known?
+    │
+    ├▶  YES
+    │
+    │   is current `context` representing `statics` ?
+    │   │
+    │   ├▶  YES
+    │   │
+    │   │   update through `interpolations`
+    │   │
+    │   └▶  NO  ┌─────────────────────────────────────────┐
+    │           ▼                                         │
+    │       clone the `fragment` related to `statics`     │
+    │                                                     │
+    │       retrieve the list of `paths` to update        │
+    │                                                     │
+    │       create a list of `updates`                    │
+    │                                                     │
+    │       relate `updates` to current `context`         │
+    │                                                     │
+    │       update `fragment` through `interpolations`    │
+    │                                                     │
+    │       replace `context` content with the `fragment` │
+    │                                                     │
+    └▶  NO                                                │
+                                                          │
+        create offline `fragment` with `statics`          │
+                                                          │
+        create a list of `paths` to update                │
+                                                          │
+        relate the `fragment` to `statics`                │
+                                                          │
+        relate the `paths` to `statics` ──────────────────┘
+```
+
+### Weakly referenced
+
+TODO: is an `expando` property that faster than a `WeakMap`?
+
+```
+context
+  ├▶  statics
+  └▶  updates
+
+any
+  └▶  wires
+```
+
+### Strongly referenced
+
+Template literals are _forever_, there's no reason to create extra GC pressure.
+
+TODO: is a native `Map` faster than a pair of arrays?
```

Comment:
This can change from one day to the next (and also across browsers) depending on how the js engine will interpret the code.